File: c/euler1.c
#!/usr/bin/pl -q -t euler -f

solve(Start, End, Total) :-
  write('Entering solve')
	(Start mod 3 =:= 0; Start mod 5 =:= 0),
	Start < End,
	Next is Start + 1,
	solve(Next, End, SemiTotal),
	Total is SemiTotal + Start.
solve(Start, End, Total) :-
	Start < End,
	Next is Start + 1,
	solve(Next, End, Total).
solve(N, N, N).

euler :-
	solve(1, 999, MyAnswer),
	format('~w~n', [MyAnswer]).

File: ocaml/euler1.ml
#!/usr/bin/ocaml
(* Euler1 in Ocaml *)

let rec range i j = i :: if i>j then [] else (range (i+1) j);;
let sum xs = List.fold_left (+) 0 xs;;

let myTest = fun x -> if x mod 3 = 0 || x mod 5 = 0 then true else false;;

let euler n = sum (List.filter myTest (range 0 n));;

Printf.printf "%d\n" (euler 998);;
